nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/ListHDFS.java

```diff
@@ -527,7 +527,7 @@ private PathFilter createPathFilter(final ProcessContext context) {
         return path -> {
             final boolean accepted;
             if (FILTER_FULL_PATH_VALUE.getValue().equals(filterMode)) {
-                accepted = filePattern.matcher(path.toString()).matches();
+                accepted = filePattern.matcher(Path.getPathWithoutSchemeAndAuthority(path).toString()).matches();
```

Comment:
If it's possible that this improvement may break existing user flows, then I'd like to discuss about other approaches to opt-in this.

We can provide different UX via different approaches:
1. Current approach: If existing flows regex contains schema or authority, their flow will not list files as before. Users may wonder what goes wrong. May not notice the change if they don't read docs..
2. Adding new 'Filter without Schema and Authority' property:
    - A. If we leave its default value blank and implement a custom validation to require it when filter regex is not empty, then we can make existing ListHDFS invalid. That will give user to chance to review their configuration.
    - B. If we use `false` as default value, existing flows work as is. While this improvement can be opt-in. The most safe approach, but a con is people may forget enabling this option.
3. Adding new 'Full Path (without schema and authority)' filter mode, or add new one and rename the existing one's display name to 'Full Path (include schema and authority)': this guarantees existing flows work as is, while providing easy configuration UX for new setups.

I personally prefer the option 3 above. How do you think?

Reply:
@ijokarumawak I'm in favor of option 1.  It's a path-segment search, not a full URI search, and I initially implemented it incorrectly.  Users would most likely report issues regarding the need to include the scheme and authority in the first place.  I understand it could cause issues in existing flows for those that do include the scheme and authority if we adjust the default behavior, but I would rather the default functionality of the full-path filter mode be applied only to the path.  The migration documentation is the appropriate place to document a change like this.  Do you agree that adding another mode or an option that pertains to only the existing full path filter mode would unnecessarily complicate the configuration of the processor?